packages/state-transition/src/util/slot.ts

```diff
@@ -21,7 +21,7 @@ export function computeTimeAtSlot(config: ChainConfig, slot: Slot, genesisTime:
   return genesisTime + slot * config.SECONDS_PER_SLOT;
 }
 
-export function getCurrentInterval(config: ChainConfig, secondsIntoSlot: number): number {
-  const timePerInterval = Math.floor(config.SECONDS_PER_SLOT / INTERVALS_PER_SLOT);
-  return Math.floor(secondsIntoSlot / timePerInterval);
+// Convert basis points to milliseconds into the slot
+export function getSlotComponentDuration(config: ChainConfig, basisPoints: number): number {
```

Comment:
There is known list of components during the slot, so we can make it more type safe. That will also make code more readable. 

Where instead of calling 

```ts
getSlotComponentDurationMs(config, config.PROPOSER_REORG_CUTOFF_BPS)
```

We would rather call as below, where function knows which components are supported here. Feels like this approach will also make code more readable and easy to know which components we have for slot. 

```
enum SlotComponents {
   PROPOSER_REORG_CUTOFF_BPS: 1, 
    ....
}
getSlotComponentDurationMs(config, SlotComponents.PROPOSER_REORG_CUTOFF_BPS)
```